### PR TITLE
fix phonenumber validation for services with sms_to_uk_landline permissions

### PR DESCRIPTION
**Pull request by rparke**

### CODE REVIEW

**Review by CrystalPea — COMMENTED** [review]
Hey, this mostly looks good, a few questions and suggestions in the comments.

Also there are a couple typos in the commit message: the new class is `PhoneNumber` (singular, not plural), and I think some letters were eaten in `services` towards the end of the message.

File: app/notifications/validators.py [hunk 1]
```diff
@@ -138,7 +141,18 @@ def validate_and_format_recipient(send_to, key_type, service, notification_type,
...
 
     if notification_type == SMS_TYPE:
-        international_phone_info = check_if_service_can_send_to_number(service, send_to)
+        if service.has_permission(SMS_TO_UK_LANDLINES):
+            try:
+                pn = PhoneNumber(send_to, allow_international=service.has_permission(INTERNATIONAL_SMS_TYPE))
+                return pn.get_normalised_format()
+            except InvalidPhoneError as e:
+                if e.code == e.Codes.NOT_A_UK_MOBILE:
+                    raise BadRequestError(message="Cannot send to international mobile numbers") from e
```
[the remark this answers is not in the document]
Comment (rparke) [reply app/notifications/validators.py:150]: I can't remember why we did this. There are certain tests that expect the NOT_A_UK_MOBILE message specifically, but we maybe should just check for the exception(?). Our tests seem to pass without it, so possibly just some bad code duplication.

[the remark this answers is not in the document]
Comment (CrystalPea) [reply app/notifications/validators.py:150]: Ok, good to test on local by turning off international sms for your service, and trying to send to international number - if the error displays correctly, I think we can be confident removing that special case from here ☺️

[the remark this answers is not in the document]
Comment (rparke) [reply app/notifications/validators.py:150]: Removed and tests pass

File: tests/app/v2/notifications/test_post_notifications.py [hunk 2]
```diff
@@ -765,19 +766,26 @@ def test_returns_a_429_limit_exceeded_if_rate_limit_exceeded(
     assert not deliver_mock.called
 
 
+@pytest.mark.parametrize(
+    "permissions",
+    [
+        [SMS_TYPE],
+        # [SMS_TYPE, SMS_TO_UK_LANDLINES], # needs utils changes to return the correct error message
```
[the remark this answers is not in the document]
Comment (rparke) [reply tests/app/v2/notifications/test_post_notifications.py:773]: I've rewritten the tests to include both use cases that we'd expect

File: tests/app/v2/notifications/test_post_notifications.py [hunk 3]
```diff
@@ -765,19 +766,28 @@ def test_returns_a_429_limit_exceeded_if_rate_limit_exceeded(
     assert not deliver_mock.called
 
 
+# This and test_post_sms_notification_returns_400_if_not_allowed_to_send_int_sms_with_new_validation will
```
Comment (CrystalPea) [inline tests/app/v2/notifications/test_post_notifications.py:769]: we don't need this comment anymore 🙌🏼

**Review by CrystalPea — COMMENTED** [review]
Looks good, just a comment needs removing 🙌🏼

**Review by CrystalPea — APPROVED** [review]

**Review by CrystalPea — APPROVED** [review]

**Pull request merged**